src/coreclr/jit/gentree.cpp

```diff
@@ -2707,6 +2707,14 @@ bool GenTree::Compare(GenTree* op1, GenTree* op2, bool swapOK)
     {
         return false;
     }
+    if (op1->OperIs(GT_MOD, GT_UMOD, GT_DIV, GT_UDIV))
+    {
+        if ((op1->gtFlags & (GTF_DIV_MOD_NO_BY_ZERO | GTF_DIV_MOD_NO_OVERFLOW)) !=
+            (op2->gtFlags & (GTF_DIV_MOD_NO_BY_ZERO | GTF_DIV_MOD_NO_OVERFLOW)))
+        {
+            return false;
+        }
+    }
```

Comment:
NOTE: these flags change semantic of a node so GenTree::Compare should take them into account